src/provider/mod.rs

```diff
@@ -838,49 +859,46 @@ async fn handle_stream(
     });
 
     // 4. Attempt to find hash
-    let (outboard, data) = match db.get(&hash) {
-        // We only respond to requests for collections, not individual blobs
-        Some(BlobOrCollection::Collection { outboard, data }) => (outboard, data),
-        _ => {
-            debug!("not found");
-            notify_transfer_aborted(events, connection_id, request_id);
-            write_response(&mut writer, &mut out_buffer, Res::NotFound).await?;
-            writer.finish().await?;
-
-            return Ok(());
-        }
-    };
-
-    // 5. Transfer data!
-    match transfer_collection(
-        hash,
-        &db,
-        writer,
-        &mut out_buffer,
-        &outboard,
-        &data,
-        events.clone(),
-        connection_id,
-        request_id,
-    )
-    .await
-    {
-        Ok(SentStatus::Sent) => {
-            let _ = events.send(Event::TransferCollectionCompleted {
+    match db.get(&hash) {
+        // Collection request
```

Comment:
this isn't true is it?  it can also be a lone blob being requested

Reply:
Yes. Updated the comment.